scipy/cluster/hierarchy.py

```diff
@@ -682,8 +681,8 @@ def linkage(y, method='single', metric='euclidean'):
         return _hierarchy.mst_single_linkage(y, n)
     elif method in ['complete', 'average', 'weighted', 'ward']:
         return _hierarchy.nn_chain(y, n, method_code)
-    else:
-        return _hierarchy.linkage(y, n, method_code)
+    elif method in ['centroid', 'median']:
```

Comment:
This looks a bit unnecessary to me. Besides, it might confuse branch coverage analysis.